src/atsamd/adc.c

```diff
@@ -51,7 +74,10 @@ static struct gpio_adc gpio_adc_pin_to_struct(uint8_t pin)
         if (adc_pins[chan] == pin)
             break;
     }
-#if CONFIG_MACH_SAMD21
+#if CONFIG_MACH_SAMC21
+    Adc* reg = (chan < 12 ? ADC0 : ADC1);
+    chan %= 12;
+#elif CONFIG_MACH_SAMD21
```

Comment:
A `%= 12` will result in a division in the underlying assembler, which will bloat the code.  Instead, pad the `adc_pins` array to 16 so that the existing code for CONFIG_MACH_SAMX5 can be reused.

Reply:
Clever, will do.